pkg/operator/ceph/config/monstore.go

```diff
@@ -71,7 +72,7 @@ func (m *MonStore) Get(who, option string) (string, error) {
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to get config setting %q for user %q", option, who)
 	}
-	return string(out), nil
+	return strings.TrimSpace(string(out)), nil
```

Comment:
What is this fixing? It should be a separate commit.

Reply:
Without this the ceph osd pool set command failed

Reply:
Do you have the error?

Reply:
2020-04-03 11:36:57.428631 I | exec: Error EINVAL: error parsing int value '8
2020-04-03 11:36:57.428895 E | op-object: failed to create or update object store toto. failed to create pools: failed to create object pools: failed to create metadata pools: failed to set pg_num_min on pool toto.rgw.control to 8

Reply:
It's interesting that we never caught this, please make it a separate commit since it's a different issue. Thanks.

Reply:
Can you share your CR file?

Reply:
The pool creation works fine with that parameter but not the pool set command, this is probably why you never caught it:
Here is the CR file: https://gist.github.com/ashangit/77619d7ebbff8fc104b906e9916a4951

Reply:
And here is the complete log:
```
2020-04-06 15:32:37.951382 I | op-object: creating object store test-1 in namespace rook-ceph
2020-04-06 15:32:37.959705 I | op-object: Gateway service running at None:9002
2020-04-06 15:32:37.959818 I | exec: Running command: ceph config get mon. rgw_rados_pool_pg_num_min --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/410161417
2020-04-06 15:32:38.578074 I | op-config: Get out 8
 out
2020-04-06 15:32:38.578236 I | exec: Running command: ceph osd pool get test-1.rgw.control all --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/448665812
2020-04-06 15:32:39.051246 I | exec: Error ENOENT: unrecognized pool 'test-1.rgw.control'
2020-04-06 15:32:39.051590 I | exec: Running command: ceph osd crush rule create-replicated test-1.rgw.control default rack --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/640261667
2020-04-06 15:32:40.501071 I | exec: Running command: ceph osd pool create test-1.rgw.control 8
 replicated test-1.rgw.control --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/229208358
2020-04-06 15:32:41.503653 I | exec: pool 'test-1.rgw.control' created
2020-04-06 15:32:41.503966 I | exec: Running command: ceph osd pool set test-1.rgw.control size 3 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/111100493
2020-04-06 15:32:42.560851 I | exec: set pool 360 size to 3
2020-04-06 15:32:42.561181 I | exec: Running command: ceph osd pool application enable test-1.rgw.control rook-ceph-rgw --yes-i-really-mean-it --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/712489479
2020-04-06 15:32:43.542700 I | exec: enabled application 'rook-ceph-rgw' on pool 'test-1.rgw.control'
2020-04-06 15:32:43.542906 I | cephclient: creating replicated pool test-1.rgw.control succeeded, buf: 
2020-04-06 15:32:43.543050 I | exec: Running command: ceph osd pool set test-1.rgw.control pg_num_min 8
 --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/249563089
2020-04-06 15:32:44.005018 I | exec: Error EINVAL: error parsing int value '8
2020-04-06 15:32:44.005271 E | op-object: failed to create or update object store test-1. failed to create pools: failed to create object pools: failed to create metadata pools: failed to set pg_num_min on pool "test-1.rgw.control" to "8\n": failed to set pool property pg_num_min on pool test-1.rgw.control: exit status 22
```
As you can see the pool creation works fine despite the trailing space but the pool set one failed